app/clients/cbc_proxy.py

```diff
@@ -100,28 +108,42 @@ def cancel_broadcast(
     ):
         pass
 
-    def _invoke_lambda(self, payload):
-        if not self.lambda_name:
-            current_app.logger.warning(
-                '{self.__class__.__name__} tried to send {payload} but cbc proxy aws env vars not set'
-            )
-            return
-
+    def _invoke_lambda_with_failover(self, payload):
         payload_bytes = bytes(json.dumps(payload), encoding='utf8')
+        result = self._invoke_lambda(self.lambda_name, payload_bytes)
+
+        if not result:
+            failover_result = self._invoke_lambda(self.failover_lambda_name, payload_bytes)
+            if not failover_result:
+                raise CBCProxyException(f'Lambda failed for both {self.lambda_name} and {self.failover_lambda_name}')
+
+        return result
 
+    def _invoke_lambda(self, lambda_name, payload_bytes):
+        current_app.logger.info(f"Calling lambda {lambda_name}")
         result = self._lambda_client.invoke(
-            FunctionName=self.lambda_name,
+            FunctionName=lambda_name,
             InvocationType='RequestResponse',
             Payload=payload_bytes,
         )
+        current_app.logger.info(f"Finished calling lambda {lambda_name}")
```

Comment:
is it possible to log the AWS Lambda request id here? i'm not sure if it is, but i'm keen to know what gets returend from boto3 (i've been reading https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/lambda.html#Lambda.Client.invoke, interested to see what it looks like in practice)

Reply:
i dont mind if we put some verbose logging here, deploy that, and then pare down the logs once we know what we can actually expect to be returned)

Reply:
I don't see anything about getting the lambda request ID, or at least not in the response from `invoke`. Let's leave the logging as this for the moment and then there is a chance we wish to remove the `finished calling` line in time.